docs/UserGuide.md

```diff
@@ -41,138 +31,225 @@ By : `Team SE-EDU`  &nbsp;&nbsp;&nbsp;&nbsp; Since: `Jun 2016`  &nbsp;&nbsp;&nbs
+> * Adding 'd/' will search by end date instead.
+> * Adding 'n/' will still search by name.
+> * If no keyword is entered, the list of all tasks is displayed.
+> * 't/', 'd/' and 'n/' can be combined to search by multiple fields
 > * Only full words will be matched e.g. `Han` will not match `Hans`
-> * Persons matching at least one keyword will be returned (i.e. `OR` search).
+> * tasks matching at least one keyword will be returned (i.e. `OR` search).
     e.g. `Hans` will match `Hans Bo`
 
 Examples:
 
-* `find John`<br>
-  Returns `John Doe` but not `john`
-* `find Betsy Tim John`<br>
-  Returns Any person having names `Betsy`, `Tim`, or `John`
+* `find Assignment`<br>
+  Returns `Assignment 1` but not `assignment 2`
+* `find Assignment Project Tutorial`<br>
+  Returns any task having names `Assignment`, `Project`, or `Tutorial`
+* 'find school t/' <br>
+  Returns any task with tags with 'school' in the name
 
-### 2.6. Deleting a person : `delete`
+### 2.5. Deleting a task : `delete`
 
-Deletes the specified person from the address book. Irreversible.<br>
+Deletes the specified task from the todo list. Irreversible.<br>
 Format: `delete INDEX`
 
-> Deletes the person at the specified `INDEX`. <br>
+> Deletes the task at the specified `INDEX`. <br>
 > The index refers to the index number shown in the most recent listing.<br>
 > The index **must be a positive integer** 1, 2, 3, ...
 
 Examples:
 
 * `list`<br>
   `delete 2`<br>
-  Deletes the 2nd person in the address book.
-* `find Betsy`<br>
+  Deletes the 2nd task in the todo list.
+* `find Project`<br>
   `delete 1`<br>
-  Deletes the 1st person in the results of the `find` command.
+  Deletes the 1st task in the results of the `find` command.
 
-### 2.7. Select a person : `select`
+### 2.6. Clearing all entries : `clear`
 
-Selects the person identified by the index number used in the last person listing.<br>
-Format: `select INDEX`
+Clears all entries from the todo list.<br>
+Format: `clear`
 
-> Selects the person and loads the Google search page the person at the specified `INDEX`.<br>
-> The index refers to the index number shown in the most recent listing.<br>
-> The index **must be a positive integer** 1, 2, 3, ...
+### 2.7. Exiting the program : `exit`
+
+Exits the program.<br>
+Format: `exit`
+
+### 2.8. Undo a command : `undo`
+
+Undoes previous commands by the user.<br>
+Format: `undo [NUMBER]`
+
+> Undoes the last data-mutating command inputted by the user.<br>
+> If a number is entered, undoes that ammount of previous commands instead.<br>
+> The number **must be a positive integer** 1, 2, 3, ...
 
 Examples:
 
-* `list`<br>
-  `select 2`<br>
-  Selects the 2nd person in the address book.
-* `find Betsy` <br>
-  `select 1`<br>
-  Selects the 1st person in the results of the `find` command.
+* `add Test`<br>
+  `undo`<br>
+  Undoes adding Test to the todo list.
+* `add Assignment` <br>
+  `add Project`
+  `undo 2`<br>
+  Undoes both commands.
```

Comment:
Not sure if "Undo" or "Undoes" here.. i think "Undo" is more correct. (Same for the one at line 148)